table.go

```diff
@@ -152,16 +152,16 @@ func (rt *RoutingTable) addPeer(p peer.ID, queryPeer bool) (bool, error) {
 
 	// the bucket to which the peer belongs is full. Let's try to find a peer
 	// in that bucket with a LastSuccessfulOutboundQuery value above the maximum threshold and replace it.
-	allPeers := bucket.peers()
-	for _, pc := range allPeers {
-		if time.Since(pc.LastUsefulAt) > rt.usefulnessGracePeriod {
-			// let's evict it and add the new peer
-			if bucket.remove(pc.Id) {
-				bucket.pushFront(&PeerInfo{Id: p, LastUsefulAt: lastUsefulAt, LastSuccessfulOutboundQueryAt: time.Now(),
-					dhtId: ConvertPeerID(p)})
-				rt.PeerAdded(p)
-				return true, nil
-			}
+	pid := bucket.selectFirst(func(pi *PeerInfo) bool {
```

Comment:
ot nit: shouldn't we pick the _least_ useful peer?

Reply:
@Stebalien What's the meaning of "ot nit" ?

Reply:
Sorry, "off topic nit". That is, a small suggestion that isn't even related to the current patch (just something I noticed while reviewing it).

(I'm basically saying "feel free to ignore this for now, but we should fix this in a future patch").